File: 고득점 kit/스택과큐/주식가격.cpp
#include <string>
#include <vector>
#include <stack>

using namespace std;

vector<int> solution(vector<int> prices) {
    vector<int> answer(prices.size());
    stack<int> st;
    
    for (int i = 0; i < prices.size(); ++i) {
        // 스택이 비어있지 않고 현재보다 작다면
        while(!st.empty() && prices[st.top()] > prices[i]) {
            answer[st.top()] = i - st.top();
            st.pop();
        }
        st.push(i);
    }
    
    // 비어있지 않다면
    while(!st.empty()) {
        answer[st.top()] = prices.size() - st.top() - 1;
        st.pop();
    }
    return answer;
}

File: self-taught graffiti/고득점 kit/스택과큐/주식가격.swift
import Foundation

func solution(_ prices:[Int]) -> [Int] { 
    var answer:[Int] = []
    var que = prices // 큐 로

    // 빌떄까지
    while que.count > 0 {
        var cur = que.first!
        que.removeFirst()
        var cnt:Int = 0

        for q in que { // 현재 보다 작을거일때만
            cnt += 1 // 증감
            if cur > q {
                break;
            }
        }

        answer.append(cnt)
    }
    return answer
}